app/dao/notifications_dao.py

```diff
@@ -107,14 +116,14 @@ def dao_get_last_date_template_was_used(template_id, service_id):
             return last_date_from_notifications
 
     if db.session.query(
-        FactNotificationStatus.query.filter(FactNotificationStatus.template_id == template_id).exists()
+        FactNotificationStatus.query.filter(FactNotificationStatus.template_id == template.id).exists()
     ).scalar():
         last_date = (
             db.session.query(functions.max(FactNotificationStatus.bst_date))
             .filter(
-                FactNotificationStatus.template_id == template_id,
+                FactNotificationStatus.template_id == template.id,
                 FactNotificationStatus.key_type != KEY_TYPE_TEST,
-                FactNotificationStatus.bst_date > datetime.now() - relativedelta(years=1),
+                FactNotificationStatus.bst_date > uniform_now - relativedelta(years=1),
```

Comment:
Not a change needed in this PR, but if searching `FactNotificationStatus` is quick enough I wonder if we still need this line (and therefore could revert https://github.com/alphagov/notifications-admin/pull/5101)

Reply:
It's never "quick enough" ;)

Srsly, I haven't done anything (yet) to make searching `FactNotificationStatus` any faster, so this PR probably doesn't improve the Worst Case Scenario very much.

I was considering adding an index on `(template_id, bst_date)` to accelerate this query.

Reply:
https://github.com/alphagov/notifications-api/pull/4305